src/ameba/ast/visitors/implicit_return_visitor.cr

```diff
@@ -0,0 +1,216 @@
+module Ameba::AST
+  class ImplicitReturnVisitor < BaseVisitor
+    # When greater than zero, indicates the current node's return value is used
+    @stack : Int32 = 0
+
+    def visit(node : Crystal::Expressions) : Nil
+      @rule.test(@source, node, @stack > 0)
+
+      last_idx = node.expressions.size - 1
+
+      swap_stack do |old_stack|
+        node.expressions.each_with_index do |exp, idx|
+          if exp.is_a?(Crystal::ControlExpression)
+            incr_stack { exp.accept(self) }
+            break
+          elsif idx == last_idx && old_stack > 0
+            incr_stack { exp.accept(self) }
+          else
+            exp.accept(self)
+          end
+        end
+      end
+    end
+
+    def visit(node : Crystal::Call) : Nil
+      @rule.test(@source, node, @stack > 0)
+
+      if node.block
+        incr_stack { node.obj.try &.accept(self) }
+      else
+        node.obj.try &.accept(self)
+      end
+
+      incr_stack {
```

Comment:
For multiline blocks use `do ... end`.

Reply:
IMO this could be another linter rule

Reply:
Agreed, I had it on my radar for a while, feel free to add it.